### ISSUE TITLE
Support for use in notebooks

### ISSUE THREAD

**Issue by cccs-RyanS** [SigmaHQ/sigma-cli#43]
Hello, I am wondering if you would be open to some modifications to this project. I know this is designed as a cli however my team would really like to use this project inside of notebooks to test and convert our sigma rules. The trouble is most of the functions we wish to use (such as the convert functions) have click annotations and do not return the data we need. So what I'm really asking is would you be open to changes that allow this to be used in a notebook? or do you think that there is a better way to meet this usecase?

**Comment by Res260** [comment]
Hey @cccs-RyanS ! I'm not part of the Sigma project, but I do have similar use cases. I maintain an internal, private backend and extensively test that backend with fake rules to validate that the conversion is done correctly.

You don't need to use `sigma-cli` at all for this, just `pysigma`. You can instantiate a `Backend` and call `mybackend.convert()` directly. 

For reference, here is code snipped for my PySigma Backend:

```python
from sigma.backends.qradar import QradarBackend  
mybackend = QradarBackend(qradar_environment='test')  # This argument is custom to my backend

from sigma.collection import SigmaCollection
rule = SigmaCollection.from_yaml("""                          
title: Regex rule
level: informational
id: fa2282fb-0499-4dff-9970-55dfc950e2dc
status: experimental
description: something
logsource:
  category: process_creation
  product: Microsoft Windows Security Event Log
detection:
  selection:
    Hostname|re|i: '.*?bitsadmin|exe'
  condition: selection
""")

output = mybackend.convert(rule)
```

To summarize: you need the `pySigma` library only to use it alongside other python code, plus the backend you want to use. `sigma-cli` should not be required for what you're trying to accomplish, if I understood correctly :) 

Hope this helps!

**Comment by cccs-RyanS** [comment]
Hey @Res260 thanks for the reply. That is kind of what I was thinking as well, but an additional reason I wish to use this package is so that I can reuse the analyze and check functions without having to basically re implement them in my notebooks. However if its just better to re-implement the logic in my notebooks that is fine

**Comment by Res260** [comment]
Fair. In that case, I'll let @thomaspatzke take over haha

**Comment by thomaspatzke** [comment]
Please use pySigma for this purpose. The CLI is by definition intended for the usage by an user on the command line and that's what the click framework and annotations are for. The CLI code is GPL licensed, so you can use it according to the license in other projects.
